plugin/restore/mysql/mysql.go

```diff
@@ -144,6 +152,154 @@ func getPITROldDatabaseName(database string, suffixTs int64) string {
 	return getSafeName(database, suffix)
 }
 
+// SyncArchivedBinlogFiles syncs the binlogs between the instance and `saveDir`, but exclude latest binlog
+func (r *Restore) SyncArchivedBinlogFiles(ctx context.Context, instance *api.Instance, saveDir string) error {
+	binlogFilesLocal, err := ioutil.ReadDir(saveDir)
+	if err != nil {
+		return err
+	}
+
+	binlogFilesOnServer, err := r.getBinlogFilesMetaOnServer(ctx)
+	if err != nil {
+		return err
+	}
+
+	latestBinlogFileOnServer, err := r.getLatestBinlogFileMeta(ctx)
+	if err != nil {
+		return err
+	}
+
+	// compare file sizes and names to decide which files to download
+	// downloadedIndex maintains the index of BinlogFile slice, for example:
+	// `SHOW MASTER LOGS` return ["binlog.000001", "binlog.000002"],
+	// if we had downloaded binlog.000001 and file size match, {0:struct{}{}} will be involved in downloadedIndex.
+	downloadedIndex := make(map[int]struct{})
```

Comment:
Since file name is equivalent across local and server, should the key be filename instead?